x-pack/plugins/case/server/common/models/commentable_case.ts

```diff
@@ -292,9 +292,12 @@ export class CommentableCase {
 
       return CaseResponseRt.encode({
         ...caseResponse,
-        comments: flattenCommentSavedObjects(subCaseComments.saved_objects),
+        // Do we always want to return the parent's comments? This might be too much data so I'm going to mark this as
+        // undefined
```

Comment:
Another potential optimization here is to only calculate the `totalAlerts` for the parent case if `this.subCase === undefined` aka meaning we're attaching an alert to a case or collection. It would allow us to avoid requesting all the comments for the parent in the situation that `this.subCase !== undefined`. It makes the code more complicated so maybe it's not worth it 🤷

Reply:
This is not the parent's comment but the `subCaseComments`. This response is only for the subCase. I moved them one level up so the UI will be able to render the user actions correctly when post/patch a comment.